server/limiter.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"sync/atomic"
+
+	"config-etcd/etcd"
+
+	"github.com/cloudwego/kitex/pkg/klog"
+	"github.com/cloudwego/kitex/pkg/limit"
+	"github.com/cloudwego/kitex/pkg/limiter"
+	"github.com/cloudwego/kitex/server"
+)
+
+// WithLimiter sets the limiter config from etcd configuration center.
+func WithLimiter(dest string, etcdClient etcd.Client, cfs ...etcd.CustomFunction) server.Option {
+	param, err := etcdClient.ServerConfigParam(&etcd.ConfigParamConfig{
+		Category:          limiterConfigName,
+		ServerServiceName: dest,
+	}, cfs...)
+	if err != nil {
+		panic(err)
+	}
+	key := param.Prefix + "/" + dest + "/" + limiterConfigName
```

Comment:
Consider how to make it customizable by the user.

For example, a service may be deployed into multiple clusters and each cluster may have independent configurations.

We don't need to introduce the 'cluster' but we do need to provide a way for the user to specify a different path

Reply:
That's why `config-nacos` has the `serverNameTemplate` design

Reply:
`dest` means serverName , so I think it can satisfy cluster requirements

Reply:
> `dest` means serverName , so I think it can satisfy cluster requirements

What if the server has two clusters (say, `test` cluster and `production` cluster), and we need to specify different limit config for them?